lib/evaluation/inputDescriptorFilterEvaluationHandler.ts

```diff
@@ -0,0 +1,115 @@
+import { Field, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+import Ajv from 'ajv';
+import jp from 'jsonpath';
+
+import { Status } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class InputDescriptorFilterEvaluationHandler extends AbstractEvaluationHandler {
+  public getName(): string {
+    return 'FilterEvaluation';
+  }
+
+  public handle(pd: PresentationDefinition, p: any, results: HandlerCheckResult[]): void {
+    const inputDescriptors: InputDescriptor[] = pd.input_descriptors;
+    for (const vc of p.verifiableCredential.entries()) {
+      this.iterateOverInputDescriptors(inputDescriptors, vc, results);
+    }
+  }
+
+  private iterateOverInputDescriptors(
+    inputDescriptors: InputDescriptor[],
+    vc: [number, any],
+    results: HandlerCheckResult[]
+  ): void {
+    for (const inputDescriptor of inputDescriptors.entries()) {
+      if (this.hasFields(inputDescriptor)) {
+        this.iterateOverFields(inputDescriptor, vc, results);
+      } else {
+        const payload = { result: [], valid: true };
+        results.push({
+          ...this.createResultObject(inputDescriptor[0], vc[0], payload),
+        });
+      }
+    }
+  }
+
+  private hasFields(inputDescriptor: [number, InputDescriptor]): boolean {
+    return (
+      inputDescriptor[1].constraints &&
+      inputDescriptor[1].constraints.fields &&
+      inputDescriptor[1].constraints.fields.length > 0
+    );
+  }
+
+  private iterateOverFields(
+    inputDescriptor: [number, InputDescriptor],
+    vc: [number, any],
+    results: HandlerCheckResult[]
+  ): void {
+    for (const field of inputDescriptor[1].constraints.fields) {
+      const inputField = this.extractInputField(vc[1], field);
+      if (!inputField.length) {
+        const payload = { result: [], valid: false };
+        this.createResponse(results, inputDescriptor, vc, payload, 'Input candidate failed to find jsonpath property');
+      } else if (!this.evaluateFilter(inputField[0], field)) {
+        const payload = { ['result']: { ...inputField[0] }, ['valid']: false };
+        this.createResponse(results, inputDescriptor, vc, payload, 'Input candidate failed filter evaluation');
+      } else {
+        const payload = { ['result']: { ...inputField[0] }, ['valid']: true };
+        results.push({
+          ...this.createResultObject(inputDescriptor[0], vc[0], payload),
+        });
+      }
+    }
+  }
```

Comment:
Function's cyclomatic complexity sould be reduced to lessthan 3

Reply:
done